solana/pyth2wormhole/client/src/healthcheck.rs

```diff
@@ -0,0 +1,49 @@
+use {
+    std::{
+        collections::VecDeque,
+        sync::Arc,
+    },
+    tokio::sync::Mutex,
+};
+
+lazy_static::lazy_static! {
+    pub static ref HEALTHCHECK_STATE: Arc<Mutex<HealthCheckState>> = Arc::new(Mutex::new(HealthCheckState::new(0)));
+}
+
+/// Helper structure for deciding service health
+pub struct HealthCheckState {
```

Comment:
Are we planning to make the logic here more complex (e.g.: not healthy if more than 50% failed)? Because I think the current logic could be implemented easier, just with a counter of "how many consecutive attestations have failed", a failure adds one and a success makes it 0.

Reply:
After seeing the rest of the code, I get why you have it. Then it makes sense to have it for reporting.

Reply:
I think it _could_ grow. I think your counter idea is very cool, I didn't think of that. But i think it's reasonable to assume we might want a percentage in the future and only a sliding window will be able to cut it.